src/components/DateRangePickerInput.jsx

```diff
@@ -41,6 +41,7 @@ const propTypes = forbidExtraProps({
   isEndDateFocused: PropTypes.bool,
   showClearDates: PropTypes.bool,
   disabled: PropTypes.bool,
+  selectivelyDisabled: PropTypes.string,
```

Comment:
@ljharb what do you think about instead providing two boolean props, `disableStart` and `disableEnd`?

It might make the logic a little easier for users. Rather than
```
let selectivelyDisabled;

if (condition1) {
  selectivelyDisabled = 'startDate';
}
if (condition2) {
  selectivelyDisabled = 'endDate';
}
<DatePicker selectivelyDisabled={selectivelyDisabled}/>
```

it could be

```
<DatePicker disableStart={condition1} disableEnd={condition2}/>
```

Would also simplify the other code a bit as well.

I guess a potential gotcha would be that behavior of `<DatePicker disableStart disableEnd/>` would probably need to be identical to `<DatePicker disabled/>`.